snp_motion_planning/src/planning_server.cpp

```diff
@@ -375,10 +418,38 @@ class PlanningServer
 
       // Set up composite instruction and environment
       tesseract_planning::CompositeInstruction program = createProgram(manip_info, fromMsg(req->tool_paths));
-      tesseract_environment::Commands env_cmds = createScanAdditionCommands(
-          req->mesh_filename, req->mesh_frame, get<std::vector<std::string>>(node_, TOUCH_LINKS_PARAM));
-      tesseract_environment::Environment::Ptr planner_env = env_->clone();
-      planner_env->applyCommands(env_cmds);
+
+      // Add the scan as a collision object to the environment
+      {
+        // Remove any previously added collision object
+        if (env_->getSceneGraph()->getLink(SCAN_LINK_NAME))
+          env_->applyCommand(std::make_shared<tesseract_environment::RemoveLinkCommand>(SCAN_LINK_NAME));
```

Comment:
Would be good to also add a service to enable removing the link and resetting the environment, which could be tied to the reset button in the main GUI